### PR TITLE
feat(new integration): onboard Monday cloud mode destination

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

>Onboarding Monday(Cloud mode destination)

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/monday/util.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+          };
+          break;
+        case "email":
+          columnValue = { email: properties[key], text: properties.emailText };
+          break;
+        case "location":
+          columnValue = {
+            address: properties[key],
+            lat: properties.latitude,
+            lng: properties.longitude
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/monday/util.js:92]: I don't think we have these fields in context.

File: v0/destinations/monday/util.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          };
+          break;
+        case "phone":
+          columnValue = {
+            phone: properties[key],
+            countryShortName: properties.countryShortName
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/monday/util.js:98]: Added check

File: v0/destinations/monday/util.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+ * This function is used to populate the final payload to be sent to the destination.
+ * @param {*} message
+ * @param {*} Config
+ * @param {*} processedResponse
+ * @returns
+ */
+const populatePayload = (message, Config, processedResponse) => {
+  const { boardId, groupTitle, columnToPropertyMapping } = Config;
+  const payload = {};
+  if (processedResponse.status === 200) {
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/monday/util.js]: Added

File: v0/destinations/monday/util.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/monday/util.js]: Added

File: v0/destinations/monday/transform.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  const event = get(message, "event");
+  const endpoint = ENDPOINT;
+  if (!event) {
+    throw new CustomError(
+      "[Monday]: event is not present in the input payloads",
+      400
+    );
+  }
+
+  const processedResponse = await getBoardDetails(endpoint, boardId, apiToken);
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/monday/transform.js:49]: updated

File: v0/destinations/monday/transform.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+  if (!destination.Config.apiToken) {
+    throw new CustomError("[Monday]: apiToken is a required field", 400);
+  }
+  if (!destination.Config.boardId) {
+    throw new CustomError("[Monday]: boardId is a required field", 400);
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/monday/transform.js]: Added

File: v0/destinations/monday/transform.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+const trackResponseBuilder = async (message, { Config }) => {
+  const { apiToken } = Config;
+  let boardId = getDestinationExternalID(message, "boardId");
+  if (!boardId) {
+    boardId = Config.boardId;
+  }
+  if (!boardId) {
+    throw new CustomError("Monday]: boardId is a required field", 400);
+  }
+  const event = get(message, "event");
```
Comment (ItsSudip) [inline v0/destinations/monday/transform.js:48]:
```suggestion
  const event = message.event;
```

File: v0/destinations/monday/transform.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+  const endpoint = ENDPOINT;
+  if (!event) {
+    throw new CustomError(
+      "[Monday]: event is not present in the input payloads",
+      400
+    );
+  }
+
```
Comment (ItsSudip) [inline v0/destinations/monday/transform.js:56]: We have to take a list of events from the dashboard for which we will proceed. No need to process it for all of the events.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/monday/util.js [hunk 8]
```diff
@@ -0,0 +1,242 @@
+const { isNumber } = require("lodash");
+const { httpPOST } = require("../../../adapters/network");
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const { CustomError, getDestinationExternalID } = require("../../util");
+
+// regex to validate latitude and longitude
+const regexExp = /^((\-?|\+?)?\d+(\.\d+)?),\s*((\-?|\+?)?\d+(\.\d+)?)$/gi;
```
Comment (yashasvibajpai) [inline v0/destinations/monday/util.js:9]: Where are we using this? I am not seeing any other reference.

File: __mocks__/data/monday/response.json [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+              "id": "group_title",
+              "title": "Next month"
+            }
+          ]
+        }
+      ]
+    },
+    "account_id": 13215538
+  }
+}
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply __mocks__/data/monday/response.json:206]: We are getting all data in one lookup call only.

File: v0/destinations/monday/transform.js (hunk 6 above)
Comment (ujjwal-ab) [reply v0/destinations/monday/transform.js:48]: Updated

File: v0/destinations/monday/transform.js (hunk 7 above)
Comment (ujjwal-ab) [reply v0/destinations/monday/transform.js:56]: Added it

File: v0/destinations/monday/util.js (hunk 8 above)
Comment (ujjwal-ab) [reply v0/destinations/monday/util.js:9]: Removed it, it was being used previously.

**Comment by ujjwal-ab** [comment]
Closing this PR due to branch name issue.

**Pull request closed without merging**